nipoppy/extractors/run_FC.py

```diff
@@ -0,0 +1,287 @@
+			participant_id = func_file_name.split('_')[i]
+		if 'ses-' in func_file_name.split('_')[i]:
+			session_id = func_file_name.split('_')[i].split('-')[1]
+		elif 'task-' in func_file_name.split('_')[i]:
+			task = func_file_name.split('_')[i]
+		elif 'run-' in func_file_name.split('_')[i]:
+			run = func_file_name.split('_')[i]
+		elif 'space-' in func_file_name.split('_')[i]:
+			space = func_file_name.split('_')[i]
+		elif 'res-' in func_file_name.split('_')[i]:
+			res = func_file_name.split('_')[i]
+
+	if res is not None:
+		space = f"{space}_{res}"
+
+	print(f"Running FC assessment for participant: {participant_id}...")
+	print("-"*50)
+
+	# check if the func file exists
+	if not os.path.exists(func_file):
+		print(f"func file not found: {func_file}")
+		print(f"Skipping participant: {participant_id}")
+		return
+
+	try:
+		for brain_atlas in brain_atlas_list:
+			print('******** running ' + brain_atlas)
+			### extract time series
+			if 'schaefer' in brain_atlas:
+				time_series, labels = extract_timeseries(func_file, brain_atlas, confound_strategy)
+			elif brain_atlas=='DKT':
+				if dkt_file is None:
+					print(f"DKT atlas file not provided")
+					continue
+				if not os.path.exists(dkt_file):
+					print(f"DKT atlas file not found: {dkt_file}")
+					continue
+				time_series, labels = extract_timeseries(func_file, dkt_file, confound_strategy)
+			else:
+				print(f"Brain atlas not recognized: {brain_atlas}")
+				continue
+
+			### assess FC
+			FC = assess_FC(time_series, labels, metric_list=metric_list)
+
+			## save output
+			if session_id is None:
+				folder = f"{output_dir}/FC/output/{participant_id}/"
+			else:
+				folder = f"{output_dir}/FC/output/{participant_id}/ses-{session_id}/"
+			if not os.path.exists(folder):
+				os.makedirs(folder)
+
+			# make the prefix for the output file
+			prefix = participant_id
+			if session_id is not None:
+				prefix = f"{prefix}_ses-{session_id}"
+			if task is not None:
+				prefix = f"{prefix}_{task}"
+			if run is not None:
+				prefix = f"{prefix}_{run}"
+			if space is not None:
+				prefix = f"{prefix}_{space}"
+			# if the brain atlas has "_" remove it
+			brain_atlas_name = brain_atlas.replace("_", "")
+			prefix = f"{prefix}_atlas-{brain_atlas_name}"
+
+			np.save(f"{folder}/{prefix}_FC.npy", FC)
+			print(f"Successfully completed FC assessment for participant: {participant_id}")
+	except Exception as e:
+		print(f"FC assessment failed with exceptions: {e}")
+		print(f"Failed participant: {participant_id}")
+
+	print("-"*75)
+	print("")
+
+
+if __name__ == '__main__':
+	# argparse
+	HELPTEXT = """
+	Script to run FC assessment
+	"""
+
+	parser = argparse.ArgumentParser(description=HELPTEXT)
+
+	parser.add_argument('--func_file', type=str, help='path to the BOLD nifti file')
+	# example: '["schaefer_100", "schaefer_200"]'
+	parser.add_argument('--brain_atlas_list', type=json.loads, help='list of brain atlases to use for FC assessment')
```

Comment:
Since there is an exact set of valid atlases, I would recommend using this instead:

```python
parser.add_argument(
    '--brain_atlas_list',
    type=str,
    choices=[],  # this should be the list of valid atlas names
    nargs='+',  # at least one atlas required
    required=True,
    help='list of brain atlases to use for FC assessment',
)
```

which would also make it clearer in the argparse usage string which values are allowed. Then something like `run_FC.py --brain_atlas_list atlas1 atlas2` would work and the atlases would still be stored in a list :)

Reply:
what if we want to let them not specify the list and use the default?

Reply:
You could add `default=['atlas1', 'atlas2']` in `parser.add_argument`. Ideally also update the help message to tell the users what the default is